libs/domain/feature-loaders/src/snapshot/SnapshotProposalLoader.ts

```diff
@@ -0,0 +1,212 @@
+import {
+    createGraphQLClient,
+    GraphQLClient,
+    ProgramError,
+} from "@banklessdao/util-data";
+import { InternalLoadContext, ScheduleMode } from "@shared/util-loaders";
+import {
+    Data,
+    Nested,
+    NonEmptyProperty,
+    OptionalNumberArrayOf,
+    OptionalObjectRef,
+    OptionalProperty,
+    RequiredArrayRef,
+    RequiredStringArrayOf,
+} from "@banklessdao/util-schema";
+import { DocumentNode } from "graphql";
+import gql from "graphql-tag";
+import * as t from "io-ts";
+import * as TE from "fp-ts/TaskEither";
+import { GraphQLDataLoaderBase } from "../base/GraphQLDataLoaderBase";
+import { BATCH_SIZE } from "../defaults";
+import { withMessage } from "io-ts-types";
+
+export const URL = "https://hub.snapshot.org/graphql";
+
+export const makeQUERY: (space: String) => DocumentNode = (space) => {
+    space = `"${space}"`; // So it gets parsed correctly
+    return gql`
+        query snapshotProposals($limit: Int,$cursor: Int) {
+            proposals (
+                first: $limit,
+                where: {
+                    space_in: [${space}],
```

Comment:
I think that I had a comment here about a possible injection vulnerability. Did you read that? It is possible that I somehow messed it up and it wasn't displayed :( The thing is that whenever you pass plain `string`s like this (without sanitization) it can be an attack vector. In this case this can turn into an [Injection Attack](https://owasp.org/www-community/Injection_Flaws). Maybe we can find a way around this?

Reply:
I take your point. I will either sanitize the string or add it as a variable the same as cursor and limit.

Reply:
spaces are now included as a query parameter instead of injected into the query directly